### PR TITLE
add tempest model build for x86 unix platforms

### PR DESCRIPTION
**Pull request by jakebolewski**
tempestmodel is an atomospheric dycore implementation.  this build will help in making it easier for the CliMA team to run comparative simulations on well defined test cases.

### CODE REVIEW

File: T/TempestModel/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+cp -v test/nonhydro_xz/NonHydroMountainCartesianTest ${bindir}
+cp -v test/nonhydro_xz/HydrostaticMountainCartesianTest ${bindir}
+cp -v test/nonhydro_xz/ThermalBubbleCartesianTest ${bindir}
+cp -v test/nonhydro_xz/Baroclinic3DCartesianRidgeTest ${bindir}
+cp -v test/nonhydro_xz/ScharMountainCartesianTest ${bindir}
+cp -v test/nonhydro_xz/InertialGravityCartesianXZTest ${bindir}
+
+cp -v test/shallowwater_sphere/BarotropicInstabilityTest ${bindir}
+cp -v test/shallowwater_sphere/MountainRossbyTest ${bindir}
+cp -v test/shallowwater_sphere/RossbyHaurwitzWaveTest ${bindir}
```
Comment (giordano) [inline T/TempestModel/build_tarballs.jl:54]: Fully automated installation, I see

File: T/TempestModel/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    Dependency("MPICH_jll"),
+    Dependency("NetCDF_jll"),
+    Dependency("HDF5_jll"),
+    # The following is adapted from NetCDF_jll
+    Dependency("LibCURL_jll"),
+    # The following libraries are dependencies of LibCURL_jll which is now a
+    # stdlib, but the stdlib doesn't explicitly list its dependencies
+    Dependency("LibSSH2_jll"),
+    Dependency("MbedTLS_jll", v"2.24.0"),
+    Dependency("nghttp2_jll"),
```
Comment (giordano) [inline T/TempestModel/build_tarballs.jl:106]: I'm still not convinced this is necessary.  We used to have a problem with stdlibs not being listed as dependencies, but that should have been fixed looong ago.  The only problem I'm aware of is that we get the wrong version of MbedTLS (no clue why, compat bounds in the registry look good, as far as I can tell) and that can be forced with
```suggestion
    # This is only an indirect dependency
    BuildDependency(PackageSpec(; name="MbedTLS_jll", version="2.24.0")),
```

Comment (jakebolewski) [reply T/TempestModel/build_tarballs.jl:106]: Ok i can try

File: T/TempestModel/build_tarballs.jl (hunk 1 above)
Comment (jakebolewski) [reply T/TempestModel/build_tarballs.jl:54]: I did automate the string formation to paste in here with julia lol

File: T/TempestModel/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    ExecutableProduct("ScharMountainCartesianTest", :ScharMountainCartesianTest_exe),
+    ExecutableProduct("InertialGravityCartesianXZTest", :InertialGravityCartesianXZTest_exe),
+
+    # shallow water test cases
+    ExecutableProduct("BarotropicInstabilityTest", :BarotropicInstabilityTest_exe), 
+    ExecutableProduct("MountainRossbyTest", :MountainRossbyTest_exe),
+    ExecutableProduct("RossbyHaurwitzWaveTest", :RossbyHaurwitzWaveTest_exe),
+]
+
+dependencies = Dependency[
```
Comment (giordano) [inline T/TempestModel/build_tarballs.jl:95]:
```suggestion
dependencies = [
```

File: T/TempestModel/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply T/TempestModel/build_tarballs.jl:54]: Yeah, sorry I was joking :slightly_smiling_face:

File: T/TempestModel/build_tarballs.jl (hunk 2 above)
Comment (giordano) [reply T/TempestModel/build_tarballs.jl]: Looks like that worked :slightly_smiling_face:   I'm still puzzled by the fact we get the wrong version of MbedTLS, as far as I can tell LibCURL_jll specifies correctly the compatibility bound: https://github.com/JuliaRegistries/General/blob/7392039acb21210b43cef86c7fa5f6f42213952e/L/LibCURL_jll/Compat.toml#L7-L9

Comment (jakebolewski) [reply T/TempestModel/build_tarballs.jl]: Cool!  I'll update the other package where I used the same form as well.

Comment (giordano) [reply T/TempestModel/build_tarballs.jl]: I believe I understood what's the problem!  Without doing anything special, Pkg installs `LibCURL_jll` 7.71, which doesn't have a compat with `MbedTLS_jll` and then installs the latest version of `MbedTLS_jll`, 2.28.  If I explicitly ask for LibCURL_jll 7.73 I get the right MbedTLS.  I think we should fix the compat with `MbedTLS_jll` also for previous versions of `LibCURL_jll`

**Pull request merged**